tgui/packages/tgui/interfaces/PaperSheet.js

```diff
@@ -277,78 +270,83 @@ class PaperSheetStamper extends Component {
       y: 0,
       rotate: 0,
     };
-  }
-
-  findStampPosition(e) {
-    const position = {
-      x: event.pageX,
-      y: event.pageY,
+    this.style = null;
+    this.handleMouseMove = e => {
+      const pos = this.findStampPosition(e);
+      if (!pos) { return; }
+      // center offset of stamp & rotate
+      pauseEvent(e);
+      this.setState({ x: pos[0], y: pos[1], rotate: pos[2] });
     };
-
-    const offset = {
-      left: e.target.offsetLeft,
-      top: e.target.offsetTop,
+    this.handleMouseClick = e => {
+      if (e.pageY <= 30) { return; }
+      const { act, data } = useBackend(this.context);
+      const stamp_obj = {
+        x: this.state.x, y: this.state.y, r: this.state.rotate,
+        stamp_class: this.props.stamp_class,
+        stamp_icon_state: data.stamp_icon_state,
+      };
+      act("stamp", stamp_obj);
     };
+  }
 
-    let reference = e.target.offsetParent;
-
-    while (reference) {
-      offset.left += reference.offsetLeft;
-      offset.top += reference.offsetTop;
-      reference = reference.offsetParent;
+  findStampPosition(e) {
+    let rotating;
+    const windowRef = document.querySelector('.Layout__content');
+    if (e.shiftKey) {
+      rotating = true;
     }
 
-    const pos = [
-      position.x - offset.left,
-      position.y - offset.top,
-    ];
-    const centerOffset = vecScale([121, 51], 0.5);
-    const center = vecSubtract(pos, centerOffset);
-    return center;
+    if (document.getElementById("stamp"))
+    {
+      const stamp = document.getElementById("stamp");
+      const stampHeight = stamp.clientHeight;
+      const stampWidth = stamp.clientWidth;
+
+      const currentHeight = rotating ? this.state.y : e.pageY
+      - windowRef.scrollTop - stampHeight;
```

Comment:
Ternary plus newline usage here is weird.
```suggestion
      const currentHeight = rotating
          ? this.state.y
          : e.pageY - windowRef.scrollTop - stampHeight;
```